set.mm

```diff
@@ -48555,13 +48585,27 @@ Contrast with domain (defined in ~ df-dm ).  For alternate definitions,
     CDFZGHABEHZABSHZIUATIZACJZBDJZTKZABESLTUAMUBUCUDIZTIUEUAUFTABCDNOUCUDTPQR
     $.
 
+  $( Intersection with cross product binary relation .  (Contributed by NM,
+     3-Mar-2007.)  (Revised by Mario Carneiro, 26-Apr-2015.) $)
+  brinxp2ALTV $p |- ( C ( R i^i ( A X. B ) ) D <->
+                     ( ( C e. A /\ D e. B ) /\ C R D ) ) $=
+    ( cxp cin wbr wa wcel brin ancom brxp anbi1i 3bitri ) CDEABFZGHCDEHZCDPHZIR
+    QICAJDBJIZQICDEPKQRLRSQCDABMNO $.
```

Comment:
The Conventions https://us.metamath.org/mpeuni/conventions.html has another institution beside ALT as well, the ALTV:
"Alternative (ALTV) versions. (...) Such alternatives should be temporary only, until it is decided which alternative should be used in the future. Alternative (*ALTV) theorems or definitions are usually contained in mathboxes. Their comments need not to contain "(Proof modification is discouraged.) (New usage is discouraged.)"."
I have a few ALTVs in my Mathbox and I think only one of them is undeniably superior to its default version: ~uniqsALTV . The others are my opinion which formed during my years of work with them. Perhaps the best timing to speak about ALTVs is when they come out of the box, i.e., now.
Two examples which I want you to compare are
```metamath
brresALTV $p |- ( C e. V -> ( B ( R |` A ) C <-> ( B e. A /\ B R C ) ) )
```
vs.
```metamath
brresg $p |- ( C e. V -> ( B ( R |` D ) C <-> ( B R C /\ B e. A ) ) )
```
and
```metamath
brinxp2ALTV $p |- ( C ( R i^i ( A X. B ) ) D <-> ( ( C e. A /\ D e. B ) /\ C R D ) )
```
vs.
```metamath
brinxp2 $p |- ( C ( R i^i ( A X. B ) ) D <-> ( C e. A /\ D e. B /\ C R D ) )
```metamath
(Let us forget for a moment that ~brinxp2ALTV or ~brinxp2 should be the ~brinxp , and) let us concentrate on a common use case when we use br with some restricted quantifier. In that case 
- we need to use the right side of ~brresg REVERSED (this is how we use in ~brinxp2 (!) already, C e. A and D e. B with the membership connectives e. in front, and C R D with general relation R on the back), see, e.g., the direct way of using ~df-rex in the proof of ~exanres , and
- the right side of ~brinxp2 GROUPED (like in ~brinxp2ALTV , ~brab2a , ~opabssxp , etc.), see, e.g., the direct way of using ~r2al in the proof of ~inxpss .

So I used ~brinxp2ALTV because I think we should use this one as the default version (as ~brinxp as a matter of fact). What do you think?

Reply:
Sorry, hadn't seen these are different statements. Acutally, you could directly revise brinxp2 to use the statement you propose, which I agree is better. You'd also have to adapt the proofs of all theorems using it. The easiest way is probably to temporarily keep the current brinxp2 (call it brinxp2-o everywhere) and prove it from the new brinxp2, and then in each theorem using it, do
```
MM-PA> expand brinxp2-o
MM-PA> minimize_with */except brinxp2-o
```
and finally remove brinxp2-o. Since you have to do it for all theorems using brinxp2-o, you can either do it manually or do a script inspired from e.g. `scripts/min.cmd` to make it automatically.

Reply:
I hope it´s OK now.